File: 2nd Year/2nd Semester/MPP/PROIECT-C#/Lab1/Lab1/Service/ParticipantService.cs
using System;
using System.Collections.Generic;
using Lab1.Domain;
using Lab1.Repository;

public class ParticipantService
{
    private readonly IParticipantRepository participantRepository;
    private readonly IRezultatRepository rezultatRepository;

    public ParticipantService(IParticipantRepository participantRepository, IRezultatRepository rezultatRepository)
    {
        this.participantRepository = participantRepository;
        this.rezultatRepository = rezultatRepository;
    }

    public List<Participant> GetAllParticipants()
    {
        var participants = participantRepository.FindAll();
        participants.Sort((p1, p2) => string.Compare(p1.LastName, p2.LastName, StringComparison.Ordinal));
        return participants;
    }

    public int CalculateTotalScore(Participant participant)
    {
        if (participant == null) return 0;

        int total = 0;
        foreach (var rezultat in rezultatRepository.FindAll())
        {
            if (rezultat.Participant.Id == participant.Id)
            {
                participant.SetPunctajProba(rezultat.TipProba, rezultat.Points);
                total += rezultat.Points;
            }
        }
        return total;
    }

    public Participant GetParticipantById(int id)
    {
        return participantRepository.FindOne(id);
    }
}

File: 2nd Year/2nd Semester/MPP/Triatlon-Java/src/main/java/com/example/laborator/Service/ParticipantService.java
package com.example.laborator.Service;

import com.example.laborator.Domain.Participant;
import com.example.laborator.Domain.Rezultat;
import com.example.laborator.Repository.IParticipantRepository;
import com.example.laborator.Repository.IRezultatRepository;
import com.example.laborator.Utils.Observable;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class ParticipantService extends Observable {
    private final IParticipantRepository participantRepository;
    private final IRezultatRepository rezultatRepository;

    public ParticipantService(IParticipantRepository participantRepository, IRezultatRepository rezultatRepository) {
        this.participantRepository = participantRepository;
        this.rezultatRepository = rezultatRepository;
    }

    public List<Participant> getAllParticipants() {
        List<Participant> participants = new ArrayList<>();
        participantRepository.findAll().forEach(participants::add);
        participants.sort(Comparator.comparing(Participant::getLast_name)
                .thenComparing(Participant::getFirst_name));
        return participants;
    }

    public int calculateTotalScore(Participant participant) {
        if (participant == null) return 0;

        int total = 0;
        // Get all results for this participant from the result repository
        for (Rezultat rezultat : rezultatRepository.findAll()) {
            if (rezultat.getParticipant().getId().equals(participant.getId())) {
                // Update the score in the participant's map
                participant.setPunctajProba(rezultat.getTipProba(), rezultat.getPunctaj());
                total += rezultat.getPunctaj();
            }
        }
        return total;
    }
}